pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/creator/impl/SegmentColumnarIndexCreator.java

```diff
@@ -179,126 +167,71 @@ public void init(SegmentGeneratorConfig segmentCreationSpec, SegmentIndexCreatio
-                new SingleValueSortedForwardIndexCreator(_indexDir, columnName, cardinality));
-          } else {
-            _forwardIndexCreatorMap.put(columnName,
-                new SingleValueUnsortedForwardIndexCreator(_indexDir, columnName, cardinality, _totalDocs));
-          }
-        } else {
-          _forwardIndexCreatorMap.put(columnName,
-              new MultiValueUnsortedForwardIndexCreator(_indexDir, columnName, cardinality, _totalDocs,
-                  indexCreationInfo.getTotalNumberOfEntries()));
-        }
-
-        // Initialize inverted index creator; skip creating inverted index if sorted
-        if (invertedIndexColumns.contains(columnName) && !indexCreationInfo.isSorted()) {
-          if (segmentCreationSpec.isOnHeap()) {
-            _invertedIndexCreatorMap.put(columnName,
-                new OnHeapBitmapInvertedIndexCreator(_indexDir, columnName, cardinality));
-          } else {
-            _invertedIndexCreatorMap.put(columnName,
-                new OffHeapBitmapInvertedIndexCreator(_indexDir, fieldSpec, cardinality, _totalDocs,
-                    indexCreationInfo.getTotalNumberOfEntries()));
-          }
-        }
-      } else {
-        // Create raw index
-        Preconditions.checkState(!invertedIndexColumns.contains(columnName),
-            "Cannot create inverted index for raw index column: %s", columnName);
-
-        ChunkCompressionType compressionType = getColumnCompressionType(segmentCreationSpec, fieldSpec);
-
-        // Initialize forward index creator
-        boolean deriveNumDocsPerChunk =
-            shouldDeriveNumDocsPerChunk(columnName, segmentCreationSpec.getColumnProperties());
-        int writerVersion = rawIndexWriterVersion(columnName, segmentCreationSpec.getColumnProperties());
-        if (fieldSpec.isSingleValueField()) {
-          _forwardIndexCreatorMap.put(columnName,
-              getRawIndexCreatorForSVColumn(_indexDir, compressionType, columnName, storedType, _totalDocs,
-                  indexCreationInfo.getLengthOfLongestEntry(), deriveNumDocsPerChunk, writerVersion));
-        } else {
-          _forwardIndexCreatorMap.put(columnName,
-              getRawIndexCreatorForMVColumn(_indexDir, compressionType, columnName, storedType, _totalDocs,
-                  indexCreationInfo.getMaxNumberOfMultiValueElements(), deriveNumDocsPerChunk, writerVersion,
-                  indexCreationInfo.getMaxRowLengthInBytes()));
-        }
       }
 
       if (textIndexColumns.contains(columnName)) {
         // Initialize text index creator
-        Preconditions.checkState(storedType == DataType.STRING,
+        Preconditions.checkState(fieldSpec.getDataType().getStoredType() == FieldSpec.DataType.STRING,
             "Text index is currently only supported on STRING type columns");
         _textIndexCreatorMap.put(columnName,
             new LuceneTextIndexCreator(columnName, _indexDir, true /* commitOnClose */));
```

Comment:
Any specific reason why this one is not included in the index creator interface?

Reply:
I don't follow this. When I reviewed it last time, TextIndexCreatorProvider supported both lucene text fst - https://github.com/apache/pinot/pull/7885#discussion_r768304406

Why do we have to exclude it ? The same distinction that we have today between lucene text index, lucene fst and native fst should be carried into this interface. When native fst was designed, we had agreed that the other 2 are not going to go away from the code (although users might be provided a migration path if need be). So, since they will continue to exist, TextIndexCreatorProvider should support all flavors imo.

Reply:
OK I will make this distinction in the interface